src/fsharp/FSharp.Core/Linq.fs

```diff
@@ -665,15 +636,45 @@ module LeafExpressionConverter =
     and failConvert inp =
             raise (new NotSupportedException(Printf.sprintf "Could not convert the following F# Quotation to a LINQ Expression Tree\n--------\n%A\n-------------\n" inp))
 
-    and transBinOp inp env addConvertLeft args addConvertRight (exprErasedConstructor : _ * _ -> _) =
-        match args with 
-        | [x1; x2] -> 
-            let e1 = ConvExprToLinqInContext env x1
-            let e2 = ConvExprToLinqInContext env x2
-            let e1 = if addConvertLeft  then Expression.Convert(e1, typedefof<Nullable<int>>.MakeGenericType [| e1.Type |]) |> asExpr else e1
-            let e2 = if addConvertRight then Expression.Convert(e2, typedefof<Nullable<int>>.MakeGenericType [| e2.Type |]) |> asExpr else e2
-            exprErasedConstructor(e1, e2) |> asExpr
-        | _ -> failConvert inp
+    and transUnaryOp inp env x (exprErasedConstructor: _ * _ -> _) fallback =
```

Comment:
That are these changes, and what problem are they fixing please? Thanks

Reply:
Supporting running quotations that would otherwise throw exceptions.

Reply:
Perhaps link to a few of the test cases that were failing?  Just so I can grok the changes